doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
 	tls = false;
 }
 ```
+### `negotiate(socket, options, timeout)` <!-- --> {#http.client.negotiate}
 
+Negotiates the HTTP settings with the server after a connection has been achieved. If TLS has been specified, this function instantiates the encryption tunnel with the server. `negotiate` is called internally by [`connect`](#http.client.connect) after the initial connection to the server and is not required to be called manually in most cases. Parameters are as follows:
```

Comment:
`negotiate` does not need a connection established to work. Infact depending on parameters the connection might not be complete when `negotiate` returns!

The fact it's called internally by `connect` is an implementation detail (and should be omitted).